Updatesurvivorstoaddweapon

reswanson opened the pull request:
@dutcherl @denman093 @bschugel1 

I wanted to try to understand how to get weapons into gamecontroller and I checked this in...

It just seemed the guid was too complicated for what we needed ,so I changed the numbers to start at 0,, 
with the zeroth weapon being none,  (if we wanted to implement that feature.)

I put some comments in the code as to what could be next, but I didnt understand how we were going to add the weapon to the survivor,  (the million dollar question)

thanks
Ron

reswanson committed:
geting weapons in gamecontroller, and how to add weapons to surivors

reswanson committed:
trying to add weapons to survivors

dutcherl reviewed (changes requested):
ISurvivor should have default constructor for when no Weapon is present. We can rework the code so that the parameter passed in at runtime is a Weapon type.

dutcherl commented:
I pulled this branch and there were a lot of errors so I am working on cleaning them up and will push the changes. We can review this PR at that time

Commit:
Hotfix - fixed ISurvivor classes and interface throwing errors .

ICharacter was forcing weapon accessor/mutator methods. Redudant for
ISurvivor and not applicable to IZombie.

Class definitions were being used like constructors.

Changed weaponId params to Weapon type to encapsulate Weapon object data
in ISurvivor class.
